pkg/load/load.go

```diff
@@ -205,32 +205,55 @@ type RegistryProxy struct {
 //
 // The running server and proxy container will be cleaned-up when Close() is called.
 func NewRegistryProxy(ctx context.Context, opts *ProxyOpts, dockerapi docker.APIClient, contentClient contentapi.ContentClient, logger progress.SubLogger) (*RegistryProxy, error) {
-	ctx, cancel := context.WithCancel(ctx)
 	proxyContainer, err := RunProxyImage(ctx, dockerapi, opts.ProxyImage, opts.RawManifest, opts.RawConfig)
 	if err != nil {
-		cancel()
 		return nil, err
 	}
 
-	transport := NewTransport(proxyContainer.Conn)
-	go func() {
-		// Canceling ctx will stop the transport.
-		_ = transport.Run(ctx, contentClient)
-	}()
+	// Ten 1-second retries to check for the proxy container being active.
+	var ready bool
+	for retry := 0; retry < 10; retry++ {
+		ready, err = ReadyBy(proxyContainer.Conn, time.Second)
+		if err != nil {
+			ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+
+			_ = StopProxyContainer(ctx, dockerapi, proxyContainer.ID)
+			return nil, err
+		}
+		if ready {
+			break
+		}
+	}
 
+	// We fail open to at least try the load because maybe we missed the ready byte.
+	// Otherwise, we would use docker load.
+	if !ready {
```

Comment:
I could error here, but, I figure we might as well try a docker pull anyway before going the slower tar load route.